test/integration/ddl_sharding_key_test.lua

```diff
@@ -547,3 +547,43 @@ pgroup.test_delete_secondary_idx = function(g)
         "Sharding key for space \"customers_secondary_idx_name_key\" is missed in primary index, specify bucket_id")
     t.assert_equals(result, nil)
 end
+
+pgroup.test_update_sharding_keys_cache = function(g)
+    local space_name = 'customers_name_key'
+    local sharding_key = helpers.update_sharding_keys_cache(g.cluster, space_name)
+    t.assert_equals(sharding_key, {'name'})
+
+    helpers.call_on_servers(g.cluster, {'s1-master', 's2-master'}, function(server)
+        server.net_box:call('set_sharding_key', {space_name, {'age'}})
+    end)
+    sharding_key = helpers.update_sharding_keys_cache(g.cluster, space_name)
+    t.assert_equals(sharding_key, {'age'})
+
+    -- Recover sharding key.
+    helpers.call_on_servers(g.cluster, {'s1-master', 's2-master'}, function(server)
+        server.net_box:call('set_sharding_key', {space_name, {'name'}})
+    end)
+    sharding_key = helpers.update_sharding_keys_cache(g.cluster, space_name)
+    t.assert_equals(sharding_key, {'name'})
+end
+
+pgroup.test_non_unique_index = function(g)
```

Comment:
I don't find relation of the test to the 'Support reset of cache with sharding keys' commit.

Reply:
Fixed.